search: remove stack getting started pages

TattdCodeMonkey opened the pull request:
## Summary

Removes the Search static Getting Started pages. The side nav links for these pages were already removed in another PR. This PR removes the pages and deeplinks etc.

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

TattdCodeMonkey committed:
search: remove stack getting started pages

TattdCodeMonkey commented on x-pack/solutions/search/plugins/enterprise_search/public/applications/enterprise_search_content/components/connectors/create_connector/finish_up_step.tsx:
```diff
@@ -311,9 +310,7 @@ export const FinishUpStep: React.FC<FinishUpStepProps> = ({ title }) => {
-                  onClick={() => {
-                    application.navigateToUrl(http.basePath.prepend(ELASTICSEARCH_PLUGIN.URL));
-                  }}
+                  href={docLinks.languageClients}
```
I was a bit unsure on this one, @sphilipse or @seanstory any opinions on better places to send users for this link with the Elasticsearch getting started guide being removed?

seanstory replied:
🤷 seems reasonable to me.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/312862)
* Commit: b20ced961e5c3a0fc9fde054862a85cb06f64503

### Failed CI Steps
* [Jest Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a92f-5145-47b8-8c6e-62e6f0c1d68b)
* [Jest Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a955-dc76-4c02-8e04-5fdc2d2ec11c)
* [FTR Configs #33](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a92f-5165-4ea5-a004-8b96bfc553cd)
* [FTR Configs #52](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a92f-5174-4252-b326-faaf531bba7b)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a92f-51a1-4c65-94ac-21dea963b5ff) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312862/jobs/0197a92f-51a1-4c65-94ac-21dea963b5ff/artifacts/0197a959-f4a2-4806-a0d4-c5503d8084e4) FTR Configs #<span></span>108 / Application Usage keys in the schema match the registered application IDs
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a92f-5174-4252-b326-faaf531bba7b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312862/jobs/0197a92f-5174-4252-b326-faaf531bba7b/artifacts/0197a956-fbe0-43da-8a1f-bd4e84654156) FTR Configs #<span></span>52 / Onboarding integration Redirect and save token
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a92f-5145-47b8-8c6e-62e6f0c1d68b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312862/jobs/0197a92f-5145-47b8-8c6e-62e6f0c1d68b/artifacts/0197a955-b949-427a-9024-7312ab285f2b) Jest Tests #<span></span>15 / Search search provider find omits connectors if config has connectors disabled
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a955-dc76-4c02-8e04-5fdc2d2ec11c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312862/jobs/0197a955-dc76-4c02-8e04-5fdc2d2ec11c/artifacts/0197a95a-242f-4a36-8d09-70d91782efc2) Jest Tests #<span></span>15 / Search search provider find omits connectors if config has connectors disabled
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a92f-5145-47b8-8c6e-62e6f0c1d68b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312862/jobs/0197a92f-5145-47b8-8c6e-62e6f0c1d68b/artifacts/0197a955-b948-4b0a-9090-86dcca81ce0c) Jest Tests #<span></span>15 / Search search provider find returns appropriate native flags when on cloud
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a955-dc76-4c02-8e04-5fdc2d2ec11c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312862/jobs/0197a955-dc76-4c02-8e04-5fdc2d2ec11c/artifacts/0197a95a-242f-4b3b-890d-f032c86f3e47) Jest Tests #<span></span>15 / Search search provider find returns appropriate native flags when on cloud
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a92f-5145-47b8-8c6e-62e6f0c1d68b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312862/jobs/0197a92f-5145-47b8-8c6e-62e6f0c1d68b/artifacts/0197a955-b949-47d8-877d-3907f7d2c173) Jest Tests #<span></span>15 / Search search provider find returns everything on empty string
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a955-dc76-4c02-8e04-5fdc2d2ec11c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312862/jobs/0197a955-dc76-4c02-8e04-5fdc2d2ec11c/artifacts/0197a95a-242f-4def-8aa5-95e1018e12e8) Jest Tests #<span></span>15 / Search search provider find returns everything on empty string
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312862#0197a92f-5165-4ea5-a004-8b96bfc553cd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312862/jobs/0197a92f-5165-4ea5-a004-8b96bfc553cd/artifacts/0197a956-9229-4e25-ad29-de222fad6e9f) FTR Configs #<span></span>33 / X-Pack Accessibility Tests - Group 3 Enterprise Search Accessibility Overview loads a landing page with product cards

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312862","number":312862,"commit":"b20ced961e5c3a0fc9fde054862a85cb06f64503"}],"number":312862}
buildkite-pr-comment-->

TattdCodeMonkey committed:
test: fix unit tests for getting started page removals

TattdCodeMonkey committed:
search: removing references to getting started pages & apps

TattdCodeMonkey committed:
update translations

kc13greiner reviewed (approved):
LGTM!

dmlemeshko reviewed (approved):
`x-pack/platform/test/scalability/apis/api.core.capabilities.json` changes LGTM

TattdCodeMonkey committed:
Merge remote-tracking branch 'upstream/main' into remove-search-static-getting-started-pages

elasticmachine committed:
Merge branch 'main' into remove-search-static-getting-started-pages

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/314641)
* Commit: 9d4370572b060098de26056effbda677be7c4edf

### Failed CI Steps
* [FTR Configs #54](https://buildkite.com/elastic/kibana-pull-request/builds/314641#0197c6eb-0e3d-4fc7-a9b8-0c70719d2447)
* [FTR Configs #54](https://buildkite.com/elastic/kibana-pull-request/builds/314641#0197c716-949b-490e-a3de-8f855ab3b6d0)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/314641#0197c6eb-0e3d-4fc7-a9b8-0c70719d2447) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/314641/jobs/0197c6eb-0e3d-4fc7-a9b8-0c70719d2447/artifacts/0197c716-3f37-455c-888d-4b9c6b85da2d) FTR Configs #<span></span>54 / Onboarding integration Redirect and save token
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/314641#0197c716-949b-490e-a3de-8f855ab3b6d0) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/314641/jobs/0197c716-949b-490e-a3de-8f855ab3b6d0/artifacts/0197c71e-0df4-4f3e-ae55-1526114204db) FTR Configs #<span></span>54 / Onboarding integration Redirect and save token

### History
* :broken_heart: [Build #314162](https://buildkite.com/elastic/kibana-pull-request/builds/314162) failed 2ae252edaa738298947af9044a6bbdf42f4bb52a
* :broken_heart: [Build #313241](https://buildkite.com/elastic/kibana-pull-request/builds/313241) failed f7fd14e5addb3643476582b801401017569e2c5b
* :broken_heart: [Build #312862](https://buildkite.com/elastic/kibana-pull-request/builds/312862) failed b20ced961e5c3a0fc9fde054862a85cb06f64503

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314641","number":314641,"commit":"9d4370572b060098de26056effbda677be7c4edf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314162","number":314162,"commit":"2ae252edaa738298947af9044a6bbdf42f4bb52a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/313241","number":313241,"commit":"f7fd14e5addb3643476582b801401017569e2c5b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312862","number":312862,"commit":"b20ced961e5c3a0fc9fde054862a85cb06f64503"}],"number":314641}
buildkite-pr-comment-->

TinaHeiligers reviewed (approved):
changes in application usage schema LGTM

TattdCodeMonkey committed:
test: update url used for search onboarding token

The overview page url redirects to home, which will redirect to start
when there are no indices. Updating this test to use
`/app/elasticsearch/start` to be more reliable and not encounter
redirects in Kibana.

sphilipse reviewed (approved):
This was a lot more work than I expected. Nice though!

TattdCodeMonkey commented:
@elasticmachine merge upstream

TattdCodeMonkey commented:
@elasticmachine merge upstream

elasticmachine commented:
merge conflict between base and head

TattdCodeMonkey committed:
Merge remote-tracking branch 'upstream/main' into remove-search-static-getting-started-pages
